src/mscorlib/src/System/Array.cs

```diff
@@ -1011,6 +1011,20 @@ public Object Clone()
             return ArraySortHelper<T>.Default.BinarySearch(array, index, length, value, comparer);
         }
 
+        public static T[] Clone<T>(T[] array)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            // Note: We should always return a new instance, even for empty arrays.
+
+            var result = new T[array.Length];
```

Comment:
Note that this does not actually create clone of the input array - the type of input array may not be T[].

To avoid issues like this, it would be better to implement it as a thin wrapper around the existing Array.Clone.